code/modules/reagents/chemistry/holder.dm

```diff
@@ -185,6 +186,10 @@
 
 	if(transfered_by && target_atom)
 		target_atom.add_hiddenprint(transfered_by) //log prints so admins can figure out who touched it last.
+		for(var/each_reagent in GLOB.restricted_reagents)
+			if(locate(each_reagent) in target_atom.reagents.reagent_list)
+				target_atom.reagents.fingerprint_transfer = transfered_by.ckey
+				break
```

Comment:
this is because it can be bypassed like these steps:
* inject welding fuel first
* inject any other reagent - it overrides the fingerprint
so that's why it uses `locate()`